core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts

```diff
@@ -259,10 +259,54 @@ export class LearnerDashboardPageComponent implements OnInit, OnDestroy {
       this.LEARNER_GROUP_FEATURE_IS_ENABLED = featureIsEnabled;
     });
 
+    let dashboardCollectionsDataPromise =
```

Comment:
I don't quite understand, are these changes related to the lesson cards?

Reply:
This retrieves data for the collections and explorations. Originally, they were only retrieved when someone clicked on the community lessons tab. I couldn't display them in the home tab without moving the call to this file.

Reply:
Waait, no, you don't copy-paste 40+ LOCs. If it's really needed, extract it to a function.

> Originally, they were only retrieved when someone clicked on the community lessons tab. I couldn't display them in the home tab without moving the call to this ngOnInit.

But how come it works now with the current lesson cards? I would expect the new components to replace the old ones without changing the learner-dashboard logic

Reply:
I can extract it to a function. Well, explorations and collections aren't originally used in the home tab so the Api was only called once a user clicks on the community lessons section (in setActiveSection).

Reply:
Ok, I see, this data is used in the redesigned home tab.

Then yes, please extract it to a function. Don't just copy paste such big chunks of code, anything bigger than 5 LOCs first consider extracting it to a function/variable/whatever. It's not that simple, but for starters this rule of thumb is good.

Reply:
Done